architecture/README.md

```diff
@@ -0,0 +1,190 @@
+    val flowObservable = flow.toObservable(coroutineScope)
+
+    val flowable: BaseFlowable<Int> = // someFlowable
+    val flowableSubject = flowable.toSubject(coroutineScope)
+
+    fun readValue(defaultValue: Int): Int? {
+        val currentFlowResult: ObservableResult<Int> by flowObservable
+        val currentFlowValue: Int? by  currentFlowResult
+        return currentFlowValue
+    }
+
+    fun postValue(value: Int) {
+        var currentSubjectResult: ObservableResult<Int> by flowableSubject
+        currentSubjectResult = ObservableResult.Result(value) 
+    }
+}
+```
+
+On Android both Observable and Subject are easily converted into `LiveData` objects (Subject being `MutableLiveData`), allowing lifecycle-aware binding.
+
+```kotlin
+val observable: Observable<Int>
+val liveData = observable.liveData
+
+val subject: Subject<Int>
+val mutableLiveData = subject.liveData
+
+init {
+    liveData.observe(lifeCycle, Observer { value ->
+        mutableLiveData.postValue(value)
+    })
+}
+```
+
+On iOS no such option exists.
+Instead observing works similar to `ReactiveX` solutions. Value changes can be observed using `observe(onNext: ...))`.
+Calling this returns a `Disposable` object. The caller is responsible for disposing the disposable after the object should no longer be observed using `dispose()`.
+A convenience `DisposeBag` is available to dispose multiple disposables at one. Use either `DisposeBag.add()` or `Disposable.putIn()` to add a Disposable to a DisposeBag.
+DisposeBags can be emptied using `dispose()`. To post new data to the Subject the `post()` method can be called.
+
+```kotlin
+val observable: Observable<Int>
+val subject: Subject<Int>
+
+val disposeBag = DisposeBag()
+
+init {
+    liveData.onNext{value ->
+        subject.post(value)
+    }.putIn(disposeBag)
+
+    // do Other stuff
+
+    disposeBag.dispose()
+}
+```
+
+## Navigation
+Navigation is available through a specialized `NavigatingViewModel`.
+This ViewModel takes a `Navigator` object that responds to a given `NavigationAction`.
+The Navigation action specifies the action(s) that can navigate.
+
+```kotlin
+sealed class SomeNavigationAction : NavigationAction<Nothing>(null) {
+    object ActionA : SomeNavigationAction()
+    object ActionB : SomeNavigationAction()
+}
+
+class SomeNavigatingViewModel(navigator: Navigator<SomeNavigationAction>): NavigatingViewModel<SomeNavigationAction>(navigator) {
+    
+    fun performActionA() {
+        navigator.navigate(ActionA)
+    }   
+
+    fun performActionB() {
+        navigator.navigate(ActionB)
+    }
+
+}
+```
+
+On the platform side the created Navigator should specify the `NavigationSpec` to be used for navigating.
+Multiple specs exist per platform, including all common navigation patterns within the app (both new screens and nested elements) as well as navigating to common OS screens (e.g opening the mail app).
+
+```kotlin
+// Android
+val viewModel = SomeNavigatingViewModel(
+    Navigator { action ->
```

Comment:
I think it would be possible to build something on top of this which does that. Or with minor changes. But the current code would be the building blocks for it.

Reply:
I created https://github.com/splendo/kaluga/issues/148 , please add ideas you have there @dasdranagon espc if you have an API in mind.